src/disjoint_mut.rs

```diff
@@ -48,6 +48,40 @@ pub struct DisjointMut<T: ?Sized + AsMutPtr> {
     inner: UnsafeCell<T>,
 }
 
+/// SAFETY: If `T: `[`Send`], then sending [`DisjointMut`]`<T>` across threads is safe.
+/// There is no non-[`Sync`] state that is left on another thread
+/// when [`DisjointMut`] gets sent to another thread.
+/// When `&`[`DisjointMut`] gets sent to another thread,
+/// that's a different story and justified below for the `unsafe impl Sync`.
+unsafe impl<T: ?Sized + AsMutPtr + Send> Send for DisjointMut<T> {}
+
+/// SAFETY: [`DisjointMut`] only provides disjoint mutable access
+/// to `T`'s elements through a shared `&`[`DisjointMut`]`<T>` reference.
+/// Thus, sharing/[`Send`]ing a `&`[`DisjointMut`]`<T>` across threads is safe.
+///
+/// More precisely, `&`[`Self`] has the two core methods of
+/// [`Self::index`] and [`Self::index_mut`]
+/// that provide disjoint immutable and mutable access to `T`'s elements.
```

Comment:
I'd like to make it very explicit that the safety of this Sync impl depends on correct usage of `index_mut` which is not marked as unsafe and requires safe usage. That's mentioned throughout this but can you boil it down to a sentence at the top in addition to the long explanation?

Reply:
How's it now?